### PR TITLE
Fix is_sqlite_stat define and other warnings

### PR DESCRIPTION
**Pull request by adizaimi**
Signed-off-by: Adi Zaimi <azaimi@bloomberg.net>

### CODE REVIEW

**Review by roborivers — APPROVED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c7e2c1d6/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c7e2c1d6/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2807'>Resubmit</a></p>

File: db/sql.h [hunk 1]
```diff
@@ -1123,8 +1123,7 @@ extern int gbl_master_swing_osql_verbose;
 extern int gbl_master_swing_sock_restart_sleep;
 
 /* takes care of both stat1 and stat2 */
-#define is_sqlite_stat(x)                                                      \
-    strncmp((x), "sqlite_stat", sizeof("sqlite_stat") - 1) == 0
+#define is_sqlite_stat(x) (strncmp((x), "sqlite_stat", strlen("sqlite_stat")) == 0)
```
Comment (riverszhang89) [inline db/sql.h:1126]: some compilers may not be able to optimize strlen away.

Comment (adizaimi) [reply db/sql.h]: ok, I put that part back. thanks

**Review by riverszhang89 — APPROVED** [review]

**Review by roborivers — APPROVED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: Success &#10003;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: Success &#10003;.</p><p><a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c7e2c1d6/detail.txt'>Detail</a> | <a href='http://comdb2.s3-website-us-east-1.amazonaws.com/tests/c7e2c1d6/logs.tar.gz'>Log</a> | <a href='http://ec2-34-202-216-70.compute-1.amazonaws.com/restricted/index.php?id=2807'>Resubmit</a></p>

**Pull request merged**